chall07/edramire.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func logrequest(start time.Time, req string, res string, status int) {
+	duration := time.Since(start)
+	fmt.Printf("%d ms - GET %s\n", duration.Nanoseconds()/1000000, req)
+	if len(res) > 0 {
+		fmt.Printf("\t %d: %s\n", status, res)
+	}
+}
+
+func getrequest(request string) (string, int) {
+	res, err := http.Get(request)
+	if err != nil {
+		log.Fatal(err)
+	}
+	stringBody, err := ioutil.ReadAll(res.Body)
+	res.Body.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return string(stringBody), res.StatusCode
+}
+
+func main() {
+	home := "https://chall07.42madrid.com/"
+	start := time.Now()
+	logrequest(start, home, "", 0)
+	answer, status := getrequest(home)
+	if status != 200 {
+		log.Fatal("Wrong response from server")
+	}
+	logrequest(start, home, answer, status)
+	values := strings.Split(answer[0:strings.Index(answer, " ")], ",")
+	r, _ := strconv.Atoi(values[1][2:])
+	g, _ := strconv.Atoi(values[2][2:])
+	b, _ := strconv.Atoi(values[3][2:])
```

Comment:
A most simple way of doing this would have been by using [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, red, green, blue int
	if _, err := fmt.Sscanf(answer, "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &red, &green, &blue); err != nil {
		log.Fatal(err)
	}
```